pkg/cmd/cli/install/install.go

```diff
@@ -0,0 +1,142 @@
+)
+
+const defaultImage = "gcr.io/heptio/velero:latest"
+
+type InstallOptions struct {
+	Namespace                 string
+	DeploymentName            string
+	Image                     string
+	BucketName                string
+	Prefix                    string
+	BackupStorageProviderName string
+	RestoreOnly               bool
+	LogLevel                  string
+	ResticTimeout             time.Duration
+	SecretName                string
+}
+
+// Args:
+//   secret
+//   plugin-dir
+//   basically all the server args?
+// Flags
+//   dry-run (yaml dump)
+
+func (o *InstallOptions) BindFlags(flags *pflag.FlagSet) {
+	// TODO Send this string down into the deployment
+	flags.StringVar(&o.DeploymentName, "deploy-name", "", "name to apply to the Velero deployment")
+	flags.StringVar(&o.BucketName, "bucket-name", "", "bucket name in which to store backups")
+	flags.StringVar(&o.Prefix, "prefix", "", "prefix under the bucket in which to store backups")
+	flags.StringVar(&o.BackupStorageProviderName, "backup-provider", "", "provider name for backup storage")
+	flags.StringVar(&o.Image, "image", "", "image to use for the Velero server deployment")
+	flags.BoolVar(&o.RestoreOnly, "restore-only", false, "run the server in restore-only mode")
+}
+
+func NewInstallOptions() *InstallOptions {
+	return &InstallOptions{}
+}
+
+func NewCommand(f client.Factory) *cobra.Command {
+	o := NewInstallOptions()
+	c := &cobra.Command{
+		Use:   "install",
+		Short: "Install Velero",
+		Long:  "Install Velero",
+		Run: func(c *cobra.Command, args []string) {
+			o.Namespace = c.Flag("namespace").Value.String()
+			cmd.CheckError(o.Complete(args, f))
+			cmd.CheckError(o.Run(c))
+		},
+	}
+
+	o.BindFlags(c.Flags())
+	output.BindFlags(c.Flags())
+	output.ClearOutputFlagDefault(c)
+
+	return c
+}
+
+func (o *InstallOptions) Run(c *cobra.Command) error {
+
+	//TODO: Extract this logic into a function inside of the install package that anyone could use.
+	var resources []runtime.Object
+
+	crds := install.CRDs()
+	for _, crd := range crds {
+		resources = append(resources, crd)
+	}
+
+	ns := install.Namespace(o.Namespace)
+	resources = append(resources, ns)
+
+	crb := install.ClusterRoleBinding(o.Namespace)
+	resources = append(resources, crb)
+
+	sa := install.ServiceAccount(o.Namespace)
+	resources = append(resources, sa)
+
+	// TODO: pass config down.
+	bsl := install.BackupStorageLocation(o.Namespace, o.BackupStorageProviderName, o.BucketName, o.Prefix, nil)
+	resources = append(resources, bsl)
+
+	deploy := install.Deployment(o.Namespace,
+		install.WithImage(o.Image),
+	)
+	resources = append(resources, deploy)
+
+	// TODO: Wrap the resources in a list instead o just printing with separators
+	for _, r := range resources {
+		fmt.Println("---")
+		if _, err := output.PrintWithFormat(c, r); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *InstallOptions) Complete(args []string, f client.Factory) error {
+	if o.Namespace == "" {
```

Comment:
rather than defaulting here, I'd instantiate the options struct with default values for each relevant field. Then, when the flags are bound, you can use the field itself as the default value for the flag, and it'll be obvious to the user what the default values are.